### DISCUSSION TITLE
Can email auth templates include results of functions?

### DISCUSSION THREAD

**Discussion by Aging-Developer** [supabase/supabase#20406] [Questions] [answered]
Hi Everyone,

I have a mobile app project that uses email authentication. and need to implement a "forget password" feature, however I didnt what use the standard method of redirecting to a webpage via the inbuild email.  

Instead i wanted to send an email that has a random 6 digit code that expires in a period of time, which they have to key into the app, before they can progress to a change password screen in the app.

my question is though:

Can the inbuilt email templates be edited in a way where i can passing in a parameter, or the template invoking a function to get the 6 digit code?

Is this a sound method for keeping it all in the app?  

Thanks in advance for response.

**Comment by GaryAustin1** [comment, marked as answer]
No you can't call functions.
There is the 6 digit .{{Token}} already available that can be used with signInWithOtp and then verifyOtp when the user enters the code.
